mobile/status_request_log.go

```diff
@@ -0,0 +1,89 @@
+package statusgo
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"runtime"
+	"runtime/debug"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/logutils/requestlog"
+)
+
+func getFunctionName(fn any) string {
+	return runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name()
+}
+
+func getShortFunctionName(fn any) string {
+	fullName := getFunctionName(fn)
+	parts := strings.Split(fullName, ".")
+	return parts[len(parts)-1]
+}
+
+// logAndCall logs request call details and executes the fn function if logging is enabled
+func logAndCall(fn any, params ...any) any {
+	defer func() {
+		if r := recover(); r != nil {
+			// we're not sure if request logging is enabled here, so we log it use default logger
+			log.Error("panic found in logAndCall", "error", r, "stacktrace", string(debug.Stack()))
+			panic(r)
+		}
```

Comment:
Can you please try to write a test for this as well?
I'm sure that you checked that it works as expected, but to prennt accidentally breaking it in the future.

At first I thought it's not possible, but then I saw you did `mockLogger` in another test, which looks very nice 👍 
And thank you for the tests in general, I'm very happy to see them ❤️